deb/ch3/spark-intro/notebooks/01_intro.ipynb

```diff
@@ -0,0 +1,261 @@
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "metadata": {},
+   "source": [
+    "pip install pyspark"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## SparkContext: how you interact with your data\n",
+    "Now that we have Pyspark rinstalled, let's restart the kernel for this notebook and figure out how to actually use it. Our first goal is to be able to load in our data. Becaues Spark is designed to work with data in HDFS and other big data storage tools as well as local data, it has its own set of data interface objects. As A SparkContext is the most basic of these for connecting to your data. In larger scale operations, a SparkContext could be configured to work across clusters or other cloud computing setups but for now, we're just using the little local cluster that is the machine that is running this notebook. To do this, we create a SparkContext object with the `'local[*]'` options. The `[*]` allows this SparkContext to have access to all local cores; you can manully set this number lower if you would like to limit the context. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pyspark\n",
+    "sc = pyspark.SparkContext('local[*]')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Resilient Distributed Datasets (RDD)\n",
+    "Now that we have a SparkContext that is ready to interact with our data, what happens to that data? The main data abstraction for Spark is the Resilient Distributed Dataset or RDD. The distributed notion of the name allows Spark to spread the data out and work on it in parallel. The resiliency comes from the redundant nature of the distrubutes similar to what was seen in HDFS. In our little local example, we won't be leveraging all of this power but we can start by taking a simple Python data structure and parallizing it into an RDD with the following code:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "ParallelCollectionRDD[0] at readRDDFromFile at PythonRDD.scala:262\n"
+     ]
+    }
+   ],
+   "source": [
+    "list_of_arrivals = [\n",
+    "    (\"PDX\", 1),\n",
+    "    (\"LAX\", 5),\n",
+    "    (\"DEN\", 3),\n",
+    "    (\"PDX\", 2),\n",
+    "    (\"JFK\", 9),\n",
+    "    (\"DEN\", 5),\n",
+    "    (\"PDX\", 7),\n",
+    "    (\"JFK\", 10),\n",
+    "]\n",
+    "arrivals_rdd = sc.parallelize(list_of_arrivals)\n",
+    "print(arrivals_rdd)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Great! We have an RDD object! But how do we see what is in our RDD? In order to avoid unecessary computation or memory usage, Spark uses **lazy evaluation**. \n",
```

Comment:
```suggestion
    "Great! We have an RDD object! But how do we see what is in our RDD? In order to avoid unnecessary computation or memory usage, Spark uses **lazy evaluation**. \n",
```

Reply:
I know you say nothing is computed before it's needed in the next line, but I guess some people might miss that lazy evaluation is when an evaluation is done as needed. Do you think we need to explicitly state what lazy evaluation is?